pkg/await/await.go

```diff
@@ -333,18 +347,34 @@ func Deletion(
 	if awaiter, exists := awaiters[id]; exists && awaiter.awaitDeletion != nil {
 		waitErr = awaiter.awaitDeletion(ctx, clientForResource, name)
 	} else {
-		objectMissing := func(d *unstructured.Unstructured, err error) error {
-			if is404(err) {
-				return nil
-			} else if err != nil {
-				glog.V(3).Infof("Received error deleting object '%s': %#v", d.GetName(), err)
-				return err
+		_ = host.LogStatus(ctx, diag.Info, urn, fmt.Sprintf("Waiting for deletion of %s '%s'", id, name))
+
+		for {
+			select {
+			case event, ok := <-watcher.ResultChan():
+				if !ok {
+					return fmt.Errorf("Timed out waiting for deletion of %s '%s'", id, name)
+				}
+
+				switch event.Type {
+				case watch.Deleted:
+					return nil
+				case watch.Error:
+					resource, _ := clientForResource.Get(name, metav1.GetOptions{})
+					return &initializationError{
+						object:    resource,
+						subErrors: []string{errors.FromObject(event.Object).Error()},
+					}
+				}
+			case <-ctx.Done(): // Handle user cancellation during watch for deletion.
+				watcher.Stop()
+				glog.V(3).Infof("Received error deleting object '%s': %#v", id, err)
+				resource, _ := clientForResource.Get(name, metav1.GetOptions{})
```

Comment:
For the error here, we have a function `is404` -- it seems to me that if we get a 404, then we can actually register this as a success, since the object was deleted.